src/Formats/FormatFactory.cpp

```diff
@@ -314,6 +314,10 @@ OutputFormatPtr FormatFactory::getOutputFormatParallelIfPossible(
 
     auto format_settings = _format_settings ? *_format_settings : getFormatSettings(context);
 
+    /// If we're handling MySQL protocol connection right now then MySQLWire is only allowed output format.
+    if (format_settings.mysql_wire.sequence_id && (name != "MySQLWire"))
```

Comment:
Ok but... maybe move this check to somewhere near MySQLHandler?
We can check for existence of FORMAT clause in AST and throw exception if any format is specified.

Reply:
Makes sense. I'll look for a good place to unify both checks.